source/Octopus.Tentacle.Client/TentacleClient.cs

```diff
@@ -109,23 +125,34 @@ public async Task<UploadResult> UploadFile(string fileName, string path, DataStr
         {
             var operationMetricsBuilder = ClientOperationMetricsBuilder.Start();
 
-            try
+            DataStream DownloadFileAction(CancellationToken ct)
             {
-                var dataStream = await rpcCallExecutor.ExecuteWithRetries(
-                    RpcCall.Create<IClientFileTransferService>(nameof(IClientFileTransferService.DownloadFile)),
-                    ct =>
-                    {
-                        logger.Info($"Beginning download of {Path.GetFileName(remotePath)} from Tentacle");
-                        var result = fileTransferServiceV1.DownloadFile(remotePath, new HalibutProxyRequestOptions(ct));
-                        logger.Info("Download complete");
-                        return result;
-                    },
-                    logger,
-                    abandonActionOnCancellation: false,
-                    operationMetricsBuilder,
-                    cancellationToken).ConfigureAwait(false);
+                logger.Info($"Beginning download of {Path.GetFileName(remotePath)} from Tentacle");
+                var result = fileTransferServiceV1.DownloadFile(remotePath, new HalibutProxyRequestOptions(ct));
+                logger.Info("Download complete");
+                return result;
+            }
 
-                return (DataStream?)dataStream;
+            try
+            {
+                if (rpcRetrySettings.RetriesEnabled)
+                {
+                    return await rpcCallExecutor.ExecuteWithRetries(
+                        RpcCall.Create<IClientFileTransferService>(nameof(IClientFileTransferService.DownloadFile)),
+                        DownloadFileAction,
+                        logger,
+                        abandonActionOnCancellation: false,
+                        operationMetricsBuilder,
+                        cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    return rpcCallExecutor.Execute(
```

Comment:
Can we explicitly set abandonActionOnCancellation: false, to make it clear what the behaviour is